json_schema/1.0.0.schema.json

```diff
@@ -0,0 +1,558 @@
+                ">= 25.0.219"
+              ]
+            }
+          ]
+        },
+        "allowed": {
+          "description": "Whether or not this dependency is allowed. Used with conditions.",
+          "$ref": "#/definitions/condition"
+        },
+        "required": {
+          "description": "Whether or not this dependency is required.",
+          "oneOf": [
+            {
+              "type": "boolean"
+            },
+            {
+              "$ref": "#/definitions/condition"
+            }
+          ]
+        }
+      }
+    },
+    "condition": {
+      "type": "object",
+      "description": "Condition match",
+      "additionalProperties": false,
+      "properties": {
+        "environment": {
+          "type": "object",
+          "description": "Minecraft Game Environment",
+          "additionalProperties": false,
+          "properties": {
+            "client": {
+              "type": "boolean",
+              "description": "Client-side"
+            },
+            "server": {
+              "type": "boolean",
+              "description": "Server-side"
+            }
+          }
+        }
+      }
+    },
+    "installation": {
+      "type": "object",
+      "description": "Installation information",
+      "additionalProperties": false,
+      "required": [
+        "method"
+      ],
+      "properties": {
+        "method": {
+          "type": "string",
+          "description": "The installation method to use.",
+          "enum": [
+            "placeInDirectory",
+            "versionJsonInstall",
+            "runInstaller",
+            "classpathLibrary",
+            "instanceInstall",
+            "jarMod",
+            "other"
+          ]
+        },
+        "directory": true,
+        "versionJson": true
+      },
+      "oneOf": [
+        {
+          "properties": {
+            "method": {
+              "const": "placeInDirectory"
+            },
+            "directory": {
+              "type": "string",
+              "description": "The directory to place this in."
+            }
+          },
+          "required": [
+            "directory"
+          ]
+        },
+        {
+          "properties": {
+            "method": {
+              "const": "classpathLibrary"
+            },
+            "versionJson": {
+              "type": "object",
+              "description": "The library as written in version JSON format."
+            }
+          }
+        },
+        {
+          "properties": {
+            "method": {
+              "enum": [
+                "versionJsonInstall",
+                "runInstaller",
+                "instanceInstall",
+                "jarMod",
+                "other"
+              ]
+            },
+            "directory": false,
+            "versionJson": false
+          }
+        }
+      ]
+    }
+  },
+  "required": [
+    "schemaVersion",
+    "id",
+    "name"
+  ],
+  "properties": {
+    "$schema": true,
```

Comment:
Same as here: https://github.com/modip-org/spec/pull/1/files#r499149675

I’m not sure if this field should be required at all, since the `schemaVersion` field already defines which schema is to be used.

Reply:
`$schema` is separate from `schemaVersion` - `$schema` is the JSON Schema URL, whereas `schemaVersion` is the version of the MODIP spec. The field is defined here because I have `additionalProperties` set to `false`, so this allows you to include the `$schema` field.

Reply:
But why should someone specify the JSON schema? Either I misunderstand the purpose of this field or it does not have any purpose. The modip spec already specifies the JSON schema it was written for, and since a project spec is only validated against the modip spec and not a JSON schema (which would not even work).

Reply:
You can see the reason why I included it in `format_example.modip.json`. I have the `$schema` field there for validation purposes. `additionalProperties` is set to false in the 1.0.0 JSON Schema, so if the `$schema` field was not listed under properties, then the example would technically be invalid. 

But it isn't really a necessary field and it's not required. I could probably get rid of `"additionalProperties": false`, then there would be no issue.